skin/frontend/base/default/js/oyst/oneclick.js

```diff
@@ -174,3 +175,60 @@ function oneClickButtonPickToFirstAddToCartButtons() {
         prependChild(addToCartButtons, oystOneClickButton);
     });
 }
+
+/**
+ * Retrieve custom options value.
+ *
+ * @returns {Array}
+ */
+function configurableProductCustomOptions() {
+    var customOptions = document.getElementsByClassName("product-custom-option");
+    var inputType;
+    var name;
+    var optionId;
+    var value;
+    var options = [];
+
+    for (var i = 0; i < customOptions.length; i++) {
+        if (customOptions[i].hasAttribute("name")) {
+            inputType = customOptions[i].getAttribute("type");
+            name = customOptions[i].getAttribute("name");
+            value = customOptions[i].value;
+            optionId = name.replace(/\D/g,'');
+            var customOption = {
+                optionId: optionId,
+                value: value
+            };
+
+            if ((inputType === "checkbox" || inputType === "radio") && customOptions[i].checked && value) {
```

Comment:
invert the comparaison (defensive programming)
```"checkbox" === inputType```

do same with other if in this function